versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDbStore.java

```diff
@@ -0,0 +1,247 @@
+   * no effect.
+   */
+  @Override
+  public void close() {
+    if (null != mongoClient) {
+      mongoClient.close();
+    }
+  }
+
+  @Override
+  public void load(LoadStep loadstep) throws ReferenceNotFoundException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <V> boolean putIfAbsent(ValueType type, V value) {
+    final MongoCollection collection = collections.get(type);
+    if (null == collection) {
+      throw new UnsupportedOperationException(String.format("Unsupported Entity type: %s", type.name()));
+    }
+
+    // TODO: Potentially rework to use updateOne with upsert and $set.
+    try (MongoCursor cursor = collection.find(Filters.eq(Store.KEY_NAME, ((HasId) value).getId())).iterator()) {
+      if (!cursor.hasNext()) {
+        collection.insertOne(value);
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  @Override
+  public <V> void put(ValueType type, V value, Optional<ConditionExpression> conditionUnAliased) {
+    Preconditions.checkArgument(type.getObjectClass().isAssignableFrom(value.getClass()),
+        "ValueType %s doesn't extend expected type %s.", value.getClass().getName(), type.getObjectClass().getName());
+
+    final MongoCollection collection = collections.get(type);
+    if (null == collection) {
+      throw new UnsupportedOperationException(String.format("Unsupported Entity type: %s", type.name()));
+    }
+
+    // TODO: Correct this so it overwrites values if already present.
+    // TODO: Throw an IllegalArgumentException (or something) if the ConditionExpression can't be satisfied.
+    collection.insertOne(value);
+  }
+
+  @Override
+  public boolean delete(ValueType type, Id id, Optional<ConditionExpression> condition) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void save(List<SaveOp<?>> ops) {
+    // TODO: Should these operations be async?
+    final ListMultimap<MongoCollection<?>, SaveOp<?>> mm = Multimaps.index(ops, l -> collections.get(l.getType()));
+    final ListMultimap<MongoCollection<?>, InsertOneModel<?>> writes =
+        Multimaps.transformValues(mm, s -> new InsertOneModel<>(s.getValue()));
+
+    if (enableTransactions) {
+      final ClientSession session = mongoClient.startSession();
+      session.withTransaction(() -> {
+        for (MongoCollection collection : writes.keySet()) {
+          Lists.partition(writes.get(collection), paginationSize).forEach(l -> collection.bulkWrite(session, l, UNORDERED));
+        }
+        return "Inserted";
+      });
+    } else {
+      for (MongoCollection collection : writes.keySet()) {
+        Lists.partition(writes.get(collection), paginationSize).forEach(l -> collection.bulkWrite(l, UNORDERED));
+      }
+    }
+  }
+
+  @Override
+  public <V> V loadSingle(ValueType valueType, Id id) {
+    final MongoCollection collection = collections.get(valueType);
+    if (null == collection) {
```

Comment:
this code is repeated multiple times. Let's extract into a private method

Reply:
Done.